autoarray/inversion/mock/mock_inversion.py

```diff
@@ -90,6 +92,14 @@ def curvature_matrix(self):
 
         return self._curvature_matrix
 
+    @property
+    def curvature_matrix_mapper_diag(self):
+
+        if self._curvature_matrix_mapper_diag is None:
+            return super()._curvature_matrix_mapper_diag
+
+        return self._curvature_matrix_mapper_diag

```

Comment:
Use @cached_property 

It doesn't exist in older versions of Python so you might want the one defined in autoconf

Reply:
You should also just make the parent property public and not bother overriding it here